python/pyarrow/tests/test_array.py

```diff
@@ -1073,6 +1073,16 @@ def test_array_from_numpy_unicode():
     assert arrow_arr.equals(expected)
 
 
+def test_array_from_masked():
+    ma = np.ma.array([1, 2, 3, 4], mask=[False, False, True, False])
+    result = pa.array(ma)
+    expected = pa.array([1, 2, None, 4])
```

Comment:
Careful. Are you sure Arrow selects the same default integer type? I don't think we should rely on that.

Reply:
Ah, yes. Fixed both numpy and arrow array to int64